synthesizer/process/src/lib.rs

```diff
@@ -65,31 +65,67 @@ use synthesizer_program::{
     StackProgram,
 };
 use synthesizer_snark::{ProvingKey, UniversalSRS, VerifyingKey};
+use tracing::{debug, info};
 
-use aleo_std::prelude::{finish, lap, timer};
-use indexmap::IndexMap;
+use aleo_std::{
+    StorageMode,
+    prelude::{finish, lap, timer},
+};
+use ledger_store::{ConsensusStore, TransactionStorage, TransactionStore};
+use lru::LruCache;
 use parking_lot::RwLock;
-use std::{collections::HashMap, sync::Arc};
+use std::{
+    collections::HashMap,
+    num::NonZeroUsize,
+    sync::{Arc, Mutex},
```

Comment:
note on the `Mutex`: we typically use the `parking_lot` primitives, which don't feature poisoning; IMO we should either stick with them, or - if we do need poisoning - include some relevant error handling for such cases

Reply:
Thank you for catching this! 4ee3a095d